analyses/cell-type-wilms-tumor-06/scripts/05_copyKAT.R

```diff
@@ -0,0 +1,135 @@
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
+# Path to the result directory
+result_dir <- file.path(module_base, "results", opts$sample_id)
+
+
+# Create directories to save the results of copykat with/without reference using opts$distance
+dir.create(file.path(result_dir,  "05_copyKAT", "noref", opts$distance), recursive = TRUE)
+dir.create(file.path(result_dir,  "05_copyKAT", "ref", opts$distance), recursive = TRUE)
+
+# define scratch directory for tempory saving the output of copykat
+scratch_dir <- file.path(module_base, "scratch", opts$sample_id)
+dir.create(scratch_dir, recursive = TRUE)
+
+# path for copykat rds output
+name_ref <- glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, ".rds")
+name_ref_full <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, name_ref)
+name_no_ref <- glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, ".rds")
+name_no_ref_full <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, name_no_ref)
+
+# path to scratch and final heatmap file to copy over
+jpeg_file <- glue::glue(opts$sample_id,"_copykat_heatmap.jpeg")
+scratch_jpeg <- file.path(scratch_dir, jpeg_file)
+output_jpeg_ref <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, "_copykat_heatmap.jpeg"))
+output_jpeg_noref <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, "_copykat_heatmap.jpeg"))
+
+# path to scratch and final .txt prediction file to copy over
+prediction_file <- glue::glue(opts$sample_id,"_copykat_prediction.txt")
+scratch_prediction <- file.path(scratch_dir, prediction_file)
+output_prediction_ref <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, "_copykat_prediction.txt"))
+output_prediction_noref <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, "_copykat_prediction.txt"))
+
+# path to scratch and final .txt CNA file to copy over
+CNA_file <- glue::glue(opts$sample_id,"_copykat_CNA_results.txt")
+scratch_CNA <- file.path(scratch_dir, CNA_file)
+output_CNA_ref <- file.path(result_dir,  "05_copyKAT", "ref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_ref_distance-", opts$distance, "_copykat_CNA_results.txt"))
+output_CNA_noref <- file.path(result_dir,  "05_copyKAT", "noref", opts$distance, glue::glue("05_copykat_",opts$sample_id,"_noref_distance-", opts$distance, "_copykat_CNA_results.txt"))
+
+
+# change working directory of the script to the scratch directory
+# this ensures copykat files get saved to the right location
+# there is no option to specify an output directory when running copykat
+setwd(scratch_dir)
```

Comment:
With full file paths defined, we shouldn't need to change directories here.

Reply:
I didn't know it, thanks!